app/models/beer.rb

```diff
@@ -1,59 +1,45 @@
+# beer model
 class Beer < ApplicationRecord
   validates_presence_of :name, :beer_type, :brand
   validates_uniqueness_of :name
   has_many :user_beers
   has_many :users, through: :user_beers
   has_many :ratings
 
-  class << self
-    def search(params)
-      beers = scope_beers(params)
-      beers = beers.where(beer_type: params["type"]) unless ["all types", ""].include?(params["type"])
-      beers = check_text(params["text"], beers) if params["text"].present?
-      beers = sort_by(beers, params["sort"]) if params["sort"].present?
-      beers = calculate_offset(beers, params) if params[:page].present?
-      beers = beers.order(updated_at: :desc).limit(24)
-      beers
-    end
-
-    def check_text(text, beers)
-      beers.where("name LIKE ?", "%#{text.downcase}%")
-    end
-
-    def scope_to_user(token)
+  scope :current_beers, (lambda do |beer, token = nil|
+    if beer == 'my beers'
       User.find_by(password_digest: token).beers
+    else
+      where(approved: true)
     end
+  end)
 
-    def sort_by(beers, criterion)
-      if criterion == "name"
-        beers.order(:name)
-      elsif criterion == "rating"
-        beers.order(average_rating: :desc)
-      elsif criterion == "abv"
-        beers.order(abv: :desc)
-      end
-    end
+  scope :beer_type, (lambda do |type|
+    where(beer_type: type) unless ['all types', ''].include?(type)
+  end)
 
-    def fetch_beer_types(params)
-      scope_beers(params)
-        .select(:beer_type)
-        .distinct.pluck(:beer_type)
-        .unshift("all types")
-    end
+  scope :beer_name, (lambda do |text|
+    where('name LIKE ?', "%#{text.downcase}%") if text.present?
+  end)
 
-    def calculate_offset(beers, params)
-      offset = (params[:page].to_i - 1) * 24
-      beers.offset(offset)
+  scope :sort_by_criterion, (lambda do |criterion|
+    case criterion
+    when 'name' then order(:name)
+    when 'rating' then order(average_rating: :desc)
+    when 'abv' then order(abv: :desc)
     end
+  end)
+
+  scope :current_page, (lambda do |page|
+    offset((page.to_i - 1) * 24) if page.present?
+  end)
 
-    private
+  scope :beer_order, (-> { order(updated_at: :desc).limit(24) })
```

Comment:
this is doing an `order` and a `limit` but the name suggests only a `order`. Better to have 2 scopes than combine these, waht if you want to order it but not put a limit? This scope doesn't allow that flexibility.